helm/vertical-pod-autoscaler-app/values.yaml

```diff
@@ -107,17 +106,25 @@ vertical-pod-autoscaler:
         controlledResources:
         - cpu
         - memory
-        maxAllowed:
-          cpu: 200m
-          memory: 3000Mi
         minAllowed:
           cpu: 100m
           memory: 250Mi
         mode: Auto
 
     extraArgs:
-      pod-recommendation-min-cpu-millicores: 15
-      pod-recommendation-min-memory-mb: 100
+      # CPU usage percentile that will be used as a base for CPU target recommendation.
+      # Doesn't affect CPU lower bound, CPU upper bound nor memory recommendations.
+      # Sets however a CPU value which matches the 95% percentile of the workload consumption
+      target-cpu-percentile: 0.95
+      # Allow the recommender to execute more requests against the api
+      kube-api-burst: 50
+      kube-api-qps: 25
+      # Double the memory during the next Pod resource update, to speed up the recovery from an OOMKilled
+      oom-bump-up-ratio: 2.0
```

Comment:
I think default is at 1.2. This is a huge difference.
For instance, with big containers like Prometheus, this will very quickly max up RAM!

Reply:
This work is about improving and stabilizing VPA.  

There are many issues that we are facing at the moment:
- The version of the VPA we are currently deploying is not supposed to be used with k8s 1.24 and only with k8s 1.25
- The VPA pods are OOMKilled constantly impacting pretty much all workloads that rely heavily on VPA
- VPA is scaling down the CPU limits to 15 millicore which is way too low and crashing important k8s components
- We don't have enough granularity to allow the VPA to make the right choices when it comes down to CPU adjustments.

The reasons for these much higher limits is because VPA is constantly killed in large clusters.

Regarding the `oom-bump-up-ratio` which is now doubled, it definitely needs testing, which is what I am currently doing.  
Again in large clusters it took 3 hours to get Prometheus started because it kept being OOMKilled. So doubling the memory each time the pods are OOMKilled will speed up the starting of Prometheus.

Reply:
I agree there's work to do on VPA stability.

BTW, did you check the OOMKilled VPAs are using the latest version of the app? Because there's already been some adjustments here: https://github.com/giantswarm/vertical-pod-autoscaler-app/pull/195


But here I warn about the parameters that will affect the global way VPA behaves.
IIRC a while ago we did consider/discuss updating this specific `oom-bump-up-ratio` parameter because of our Prometheus issues, and decided not to. I can't remember why though.

If anything goes wrong with Prometheus it's not a big deal though: it runs on MCs so it's fast to revert.
I'm more concerned about WCs: deploying a fix to all WCs takes ages.

I totally support you with these changes, just wanted to make sure you are aware of the possible impact :wink:

Reply:
I did check the adjustments on that PR but unfortunately we are still close to 90% of used memory.  
One thing we could do to decrease the memory and in general the resource consumption is to set the VPA to only calculate the recommendations for the workloads which do have a VPA CR set.  
By default it calculates the recommendation for all workloads.

Nevertheless I am in the process of testing various settings to try to find better default for various workloads. Prometheus being one of them